common/src/main/java/io/quarkus/ts/openshift/common/OpenShiftTestExtension.java

```diff
@@ -149,7 +149,14 @@ private void doBeforeAll(ExtensionContext context) throws Exception {
 
             awaitImageStreams(context, openshiftResources);
 
-            new Command("oc", "start-build", getAppMetadata(context).appName, "--from-dir=target", "--follow").runAndWait();
+            // when generating Kubernetes resources, Quarkus expects that all application files
+            // will reside in `/deployments/target`, but if we did just `oc start-build my-app --from-dir=target`,
+            // the files would end up in `/deployments`
+            // using `tar` works around that nicely, because the tarball created with this command will have
+            // a single root directory named `target`, which the S2I builder image will unpack to `/deployments`
+            new Command("tar", "czf", "app.tar.gz", "target").runAndWait();
```

Comment:
Can we assume `tar` being available on all the systems or there should be explicit note in README?

Reply:
Hmm. Good point. Or I can bring in a Java implementation of `tar`... which I really wouldn't like to do :-) I mean, `tar` is everywhere :-) But adding a "prerequisites" section to the readme is a good idea.

Reply:
+1 for "prerequisites" section to the readme
no need to pull in java-tar